### PR TITLE
Describe conformance for internal components

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/next/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/material-ui-lab/src/MonthPicker/MonthPicker.test.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        maxDate={new Date('2029-01-01')}
+        date={new Date('2019-02-02')}
+        onChange={onChangeMock}
+      />,
+    );
+
+    fireEvent.click(screen.getByText('May', { selector: 'button' }));
+
+    // @ts-expect-error firstArg is missing in sinon definitions
+    expect((onChangeMock.firstCall.firstArg as Date).getMonth()).to.equal(4); // month index starting from 0
```
Comment (oliviertassinari) [inline packages/material-ui-lab/src/MonthPicker/MonthPicker.test.tsx:61]:
```suggestion
    expect((onChangeMock.args[0][0] as Date).getMonth()).to.equal(4); // month index starting from 0
```

File: packages/material-ui-lab/src/MonthPicker/MonthPicker.tsx [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 
 export interface MonthPickerProps<TDate> {
   date: TDate | null;
+  /** Minimal selectable date */
   minDate: TDate;
+  /** Maximal selectable date */
   maxDate: TDate;
+  /** Callback fired on date change */
   onChange: PickerOnChangeFn<TDate>;
+  /** If `true` past days will be disabled */
```
Comment (oliviertassinari) [inline packages/material-ui-lab/src/MonthPicker/MonthPicker.tsx:16]: present tense, comma, leading dot, multi-line block comment:

```suggestion
  /**
   * If `true`, past days are disabled.
   * /
```

File: docs/src/pages/components/date-picker/date-picker.md [hunk 3]
```diff
@@ -74,6 +74,12 @@ For ease of use the date picker will automatically change the layout between por
 
 {{"demo": "pages/components/date-picker/StaticDatePickerLandscape.js", "bg": true}}
 
+## Internal pickers
```
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply docs/src/pages/components/date-picker/date-picker.md:77]: I think about moving this at all from the `date-picker` documentation. But I am in doubt. What do you think @oliviertassinari?

File: packages/material-ui-lab/src/ClockPicker/ClockPickerStandalone.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  > {
+  /** Controlled clock view. */
+  view?: TimePickerView;
+  /** Available views for clock. */
+  views?: TimePickerView[];
+  /** Callback fired on new change */
+  onViewChange?: (view: TimePickerView) => void;
+  /** Initially opened views. */
+  openTo?: TimePickerView;
+  /** Root element class name */
```
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply packages/material-ui-lab/src/ClockPicker/ClockPickerStandalone.tsx:20]: The problem here is that we are not spreading props except this one. This is the long story problem if you remember. 
In order to implement props spreading we need to refactor a lot.

File: packages/material-ui-lab/src/internal/pickers/hooks/usePickerState.ts [hunk 5]
```diff
@@ -92,7 +92,7 @@ export function usePickerState<TInput, TDateValue>(
       isMobileKeyboardViewOpen,
       toggleMobileKeyboardView: () => {
         if (!isMobileKeyboardViewOpen) {
-          // accept any partial input done by React.user
+          // accept any partial input done by user
```
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply packages/material-ui-lab/src/internal/pickers/hooks/usePickerState.ts:95]: Haha, I just realized that we actually don't need this line at all after v4

File: docs/src/pages/components/date-picker/date-picker.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (oliviertassinari) [reply docs/src/pages/components/date-picker/date-picker.md:77]: What do you mean by "moving this at all"?

File: packages/material-ui-lab/src/ClockPicker/ClockPickerStandalone.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (oliviertassinari) [reply packages/material-ui-lab/src/ClockPicker/ClockPickerStandalone.tsx]: ok

File: docs/src/pages/components/date-picker/date-picker.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dmtrKovalenko) [reply docs/src/pages/components/date-picker/date-picker.md:77]: Moving out from the date-picker docs. Place it somewhere else

[the remark this answers is not in the document]
Comment (oliviertassinari) [reply docs/src/pages/components/date-picker/date-picker.md:77]: Ohh, well, I think that we could solve that, in another life :). I still need to do this docs structure I said I would in our last meeting.

**Review by oliviertassinari — APPROVED** [review]

**Pull request merged**